source/physics/UniformElectricDriftField.cc

```diff
@@ -84,23 +87,18 @@ namespace nexus {
     // Set the new time and position of the drifting charge
     xyzt.set(time, position);
 
+    G4double rnd = -lifetime_ * log(G4UniformRand());
+    if (xyzt.t() > rnd) step_length = 0.;
```

Comment:
I'm not sure about this line. I've run two small Kr productions with a short lifetime with the current code and with these changes and they don't match (the current code gives the correct value of the lifetime, but this one doesn't). I have to figure that out before approving the PR.

Reply:
Solved.